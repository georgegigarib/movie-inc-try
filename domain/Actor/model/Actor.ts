export class Actor {
	constructor(
		public readonly name: string,
		public readonly id: number,
		public readonly character: string,
		public readonly profilePath: string
	) {
		this.name = name,
		this.id = id,
		this.character = character,
		this.profilePath = profilePath
	}
}
