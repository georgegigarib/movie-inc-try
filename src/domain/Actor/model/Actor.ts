import { ActorDto } from "../dtos/Dtos";

export class Actor {
	constructor(
		public readonly name: string,
		public readonly id: number,
		public readonly character: string,
		public readonly profilePath: string
	) {
	}

	static fromDto(dto: ActorDto): Actor {
        return new Actor(
            dto.name,
            dto.id,
            dto.character,
            `${process.env.EXPO_PUBLIC_API_IMAGE_BASE_URL}${dto.profile_path}`
        );
    }
}
