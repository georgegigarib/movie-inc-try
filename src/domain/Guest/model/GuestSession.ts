export class GuestSession {
    constructor(
      public readonly guestSessionId: string,
      public readonly expireDate: string
    ) {
      this.guestSessionId = guestSessionId;
      this.expireDate = expireDate;
    }
  }
  