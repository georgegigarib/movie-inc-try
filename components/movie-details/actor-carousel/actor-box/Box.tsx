import React from 'react';
import { StyleSheet, TouchableOpacity } from 'react-native';
import { ThemedText } from '@/components/ThemedText';
import { ThemedView } from '@/components/ThemedView';
import ImageWithLoader from '@/components/image-loader-box/ImageLoaderBox'
import { Actor } from '@/src/domain/Actor/model/Actor';

interface ActorBoxProps {
  actor: Actor;
}

export default function Box({ actor }: ActorBoxProps) {
  return (
    <TouchableOpacity activeOpacity={0.7}>
      <ThemedView style={styles.box}>
          <ImageWithLoader
            source={{ uri: actor.profilePath }}
            style={styles.posterContainer}
          />
        <ThemedView style={styles.names}>
          <ThemedText style={styles.title}>{actor.name}</ThemedText>
          <ThemedText style={styles.caption}>{actor.character}</ThemedText>
        </ThemedView>
      </ThemedView>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  box: {
    width: 100,
    height: 200,
    marginHorizontal: 10,
    alignItems: 'center',
  },
  names: {
    alignItems: 'center',
  },
  posterContainer: {
    width: 100,
    height: 150,
    borderRadius: 5,
    borderWidth: 1,
    borderColor: '#ddd',
    overflow: 'hidden',
    backgroundColor: 'transparent',
    elevation: 15,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 0 },
    shadowOpacity: 0.8,
  },
  title: {
    fontSize: 12,
    fontWeight: '600',
    textAlign: 'center',
  },
  caption: {
    fontSize: 10,
    fontWeight: '400',
    textAlign: 'center',
  },
});