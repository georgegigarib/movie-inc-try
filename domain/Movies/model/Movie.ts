import { Actor } from "@/domain/Actor/model/Actor";

export class Movie {
	public readonly id: number;
	public readonly title: string;
	public readonly releaseDate: string;
	public readonly voteAverage: number;
	public readonly posterPath: string;
	public readonly overview?: string;
	public readonly genres?: { id: number, name: string }[];
	public readonly actors?: Actor[];
	public readonly rating?: Number;

	constructor(
		id: number,
		title: string,
		releaseDate: string,
		voteAverage: number,
		posterPath: string,
		overview?: string,
		genres?: { id: number, name: string }[],
		actors?: Actor[],
		rating?: number
	) {
		this.id = id;
		this.title = title;
		this.releaseDate = releaseDate;
		this.voteAverage = voteAverage;
		this.posterPath = posterPath;
		this.overview = overview;
		this.genres = genres;
		this.actors = actors;
		this.rating = rating;
	}
}
