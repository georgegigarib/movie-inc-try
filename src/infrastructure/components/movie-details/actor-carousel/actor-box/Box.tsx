import React from 'react';
import { StyleSheet, TouchableOpacity } from 'react-native';
import { ThemedText } from '@/src/infrastructure/components/ThemedText';
import { ThemedView } from '@/src/infrastructure/components/ThemedView';
import ImageWithLoader from '@/src/infrastructure/components/image-loader-box/ImageLoaderBox';
import { Actor } from '@/src/domain/Actor/model/Actor';

interface ActorBoxProps {
  actor: Actor;
}

export default function Box({ actor }: ActorBoxProps) {
  return (
    <TouchableOpacity activeOpacity={0.7} testID="actor-box-button">
      <ThemedView style={styles.box} testID="actor-box">
        <ImageWithLoader
          source={{ uri: actor.profilePath }}
          style={styles.posterContainer}
          testID="actor-image"
        />
        <ThemedView style={styles.names} testID="actor-names">
          <ThemedText style={styles.title} testID="actor-name">{actor.name}</ThemedText>
          <ThemedText style={styles.caption} testID="actor-character">{actor.character}</ThemedText>
        </ThemedView>
      </ThemedView>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  box: {
    width: 100,
    height: 200,
    marginHorizontal: 10,
    alignItems: 'center',
  },
  names: {
    alignItems: 'center',
  },
  posterContainer: {
    width: 100,
    height: 150,
    borderRadius: 5,
    borderWidth: 1,
    borderColor: '#ddd',
    overflow: 'hidden',
    backgroundColor: 'transparent',
    elevation: 15,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 0 },
    shadowOpacity: 0.8,
  },
  title: {
    fontSize: 12,
    fontWeight: '600',
    textAlign: 'center',
  },
  caption: {
    fontSize: 10,
    fontWeight: '400',
    textAlign: 'center',
  },
});
