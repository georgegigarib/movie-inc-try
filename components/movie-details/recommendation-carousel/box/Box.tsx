import React from 'react';
import { StyleSheet, TouchableOpacity } from 'react-native';
import { ThemedText } from '@/components/ThemedText';
import ImageWithLoader from '@/components/image-loader-box/ImageLoaderBox';
import { Movie } from '@/src/domain/Movies/model/Movie';
import { ThemedView } from '@/components/ThemedView';

interface MovieBoxProps {
  movie: Movie;
  onPress: () => void;
}

const Box: React.FC<MovieBoxProps> = ({ movie, onPress }) => {
  return (
    <TouchableOpacity onPress={onPress} activeOpacity={0.7}>
      <ThemedView style={styles.movieBox}>
        <ImageWithLoader
          source={{ uri: movie.posterPath }}
          style={styles.posterContainer}
        />
        <ThemedView style={styles.names}>
          <ThemedText style={styles.title}>{movie.title}</ThemedText>
        </ThemedView>
      </ThemedView>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  movieBox: {
    width: 100,
    marginHorizontal: 10,
  },
  posterContainer: {
    width: 100,
    height: 150,
    borderRadius: 5,
    borderWidth: 1,
    borderColor: '#ddd',
    overflow: 'hidden',
    backgroundColor: 'transparent',
    elevation: 15,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 0 },
    shadowOpacity: 0.8,
  },
  names: {
    alignItems: 'center',
  },
  title: {
    fontSize: 12,
    fontWeight: '600',
    textAlign: 'center',
  },
});

export default Box;
