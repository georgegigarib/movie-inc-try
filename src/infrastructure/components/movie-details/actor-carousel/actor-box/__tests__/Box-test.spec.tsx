import React from 'react';
import { render } from '@testing-library/react-native';
import Box from '@/src/infrastructure/components/movie-details/actor-carousel/actor-box/Box';
import { Actor } from '@/src/domain/Actor/model/Actor';

const mockActor: Actor = new Actor(
  'John Doe',
  1,
  'Main Character',
  'https://example.com/profile.jpg'
);

describe('Box', () => {
  it('renders correctly and shows actor details', () => {
    const { getByTestId, toJSON } = render(<Box actor={mockActor} />);
    
    expect(getByTestId('actor-box-button')).toBeTruthy();
    expect(getByTestId('actor-box')).toBeTruthy();
    expect(getByTestId('actor-image')).toBeTruthy();
    expect(getByTestId('actor-names')).toBeTruthy();
    expect(getByTestId('actor-name').children[0]).toBe('John Doe');
    expect(getByTestId('actor-character').children[0]).toBe('Main Character');
    
    expect(toJSON()).toMatchSnapshot();
  });
});
