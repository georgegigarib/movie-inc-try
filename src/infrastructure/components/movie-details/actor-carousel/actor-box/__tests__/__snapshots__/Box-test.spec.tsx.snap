// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box renders correctly and shows actor details 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="actor-box-button"
>
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        {
          "alignItems": "center",
          "height": 200,
          "marginHorizontal": 10,
          "width": 100,
        },
      ]
    }
    testID="actor-box"
  >
    <View
      style={
        [
          {
            "backgroundColor": "#fff",
          },
          {
            "backgroundColor": "transparent",
            "borderColor": "#ddd",
            "borderRadius": 5,
            "borderWidth": 1,
            "elevation": 15,
            "height": 150,
            "overflow": "hidden",
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0.8,
            "width": 100,
          },
        ]
      }
      testID="actor-image"
    >
      <View
        style={
          [
            {
              "backgroundColor": "#fff",
            },
            {
              "alignItems": "center",
              "backgroundColor": "rgba(255, 255, 255, 0.8)",
              "bottom": 0,
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
          ]
        }
        testID="loading-container"
      >
        <ActivityIndicator
          color="#b6b6b6"
          size="large"
        />
      </View>
      <View
        collapsable={false}
        style={
          {
            "opacity": 0,
          }
        }
      >
        <Image
          onError={[Function]}
          onLoad={[Function]}
          onLoadStart={[Function]}
          source={
            {
              "uri": "https://example.com/profile.jpg",
            }
          }
          style={
            {
              "height": "100%",
              "width": "100%",
            }
          }
          testID="actor-image-image"
        />
      </View>
    </View>
    <View
      style={
        [
          {
            "backgroundColor": "#fff",
          },
          {
            "alignItems": "center",
          },
        ]
      }
      testID="actor-names"
    >
      <Text
        style={
          [
            {
              "color": "#11181C",
            },
            {
              "fontSize": 16,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            {
              "fontSize": 12,
              "fontWeight": "600",
              "textAlign": "center",
            },
          ]
        }
        testID="actor-name"
      >
        John Doe
      </Text>
      <Text
        style={
          [
            {
              "color": "#11181C",
            },
            {
              "fontSize": 16,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            {
              "fontSize": 10,
              "fontWeight": "400",
              "textAlign": "center",
            },
          ]
        }
        testID="actor-character"
      >
        Main Character
      </Text>
    </View>
  </View>
</View>
`;
